homeassistant/components/iaqualink/climate.py

```diff
@@ -73,6 +74,22 @@ async def async_set_hvac_mode(self, hvac_mode: HVACMode) -> None:
         else:
             _LOGGER.warning("Unknown operation mode: %s", hvac_mode)
 
+    @property                                    
+    def hvac_action(self) -> HVACAction:         
+        """Return the current HVAC action."""    
+        if self.dev.is_on is False:    
+            return HVACAction.OFF
+        if self.current_temperature is not None \
+            and float(self.dev.current_temperature) < self.target_temperature:
```

Comment:
Small suggestion, but if you restructure the if/else logic a big then you don't have to check current_temperature is not None and them in the elif current_temperature is None.

Reply:
Thanks @dcmeglio for the suggestion. I've reordered the logic to clean it up.